Makefile

```diff
@@ -126,13 +129,18 @@ vet: ## Run go vet against code.
 	$(GO_CMD) vet ./...
 
 .PHONY: test
-test: generate fmt vet ## Run tests.
-	$(GO_CMD) test $(GO_TEST_FLAGS) $(shell go list ./... | grep -v '/test/') -coverprofile cover.out
+test: generate fmt vet gotestsum ## Run tests.
+	$(GOTESTSUM) --junitfile unit.xml -- $(GO_TEST_FLAGS) $(shell go list ./... | grep -v '/test/') -coverprofile cover.out
+	echo $(ARTIFACTS)/junit.xml
+	mv unit.xml $(ARTIFACTS)/junit.xml
 
 .PHONY: test-integration
 test-integration: manifests generate fmt vet envtest ginkgo ## Run tests.
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) --arch=amd64 use $(ENVTEST_K8S_VERSION) -p path)" \
-	$(GINKGO) -v $(INTEGRATION_TARGET)
+	$(GINKGO) --junit-report=integration.xml -v $(INTEGRATION_TARGET)
+	echo $(ARTIFACTS)/junit.xml
+
+	mv integration.xml $(ARTIFACTS)/junit.xml
```

Comment:
why not just put it in $(ARTIFACTS)/junit.xml in the line above? `--junit-report`

for most of these output files like cover.out I would do this. prow can also render go cover profiles for example

Reply:
I am able to do this for gotestsum.  

But for GINKGO I am running into issues. https://github.com/onsi/ginkgo/issues/1001 is related to this.

Reply:
Ah yes, check kubernetes gingko-e2e.sh I think we set a separate flag for this

Reply:
Woah.  That was a lot of flags!

I think your hint below helped me along.